docs/src/symbol_searching.md

```diff
@@ -0,0 +1,97 @@
+# Working with Symbols
+
+Tables use `Symbol`s to represent column names.
+Often one thus wants to do various filters on `Symbol`s.
+
+Most filtering functions only accept strings, so code contains a lot of code like:
+```julia
+filter(col_name->startswith(colname, "temperature"), String.(names(df)))
+```
+
+We solve that by making functions that accept `Symbol` for all the arguments that would otherwise be strings.
+Similarly, while some functions like `isequal` and `in` have _curried_ varients, like `in(collection)` being the same as `item->in(item, collection)`,
+not all do.
+We add the missing ones.
+
+
+!!! warning "Type Piracy"
+    We type-pirate `startwith` and `endswith` to provide the single argument, and `Symbol` accepting versions.
+    This is misdemeanor type-piracy: it only turns code which currently errors into non-errors.
```

Comment:
Just namespace these inside `Wrangler` and have people do `import Wrangler.startswith` to opt into your version of the functions? No need for type piracy.

Reply:
I debated this option. It ones one of a few, others being type-piracy, and making up new names `prefixedby` and `suffixedby`

I would have said it as putting in the docs telling people to always do:
```
using Wrangingly
using Wranging: startswith, endswith
```

rather than just `using Wrangling`.

I think @nickrobinson251  convinced me that type-piracy was the least bad of many options

Reply:
yeah... i'm no longer sure type-piracy is the best option. But 🤷‍♂ 

My thinking was:
- The alternatively-named-function feels a bad compromise because they are really the exact same as startswith/endswith... so they really should be methods of those functions. 
- But introducing the name-clash with Base is also pretty un-fun for users. 
- Which leaves us with type-piracy...

Reply:
I am going to go with the type-piracy for now.
But these are good cases.
We can change it later, its version 0.1 of the package we are talking about.

I will convert this to an issue